cpp/src/parquet/column_writer.cc

```diff
@@ -2374,11 +2377,8 @@ std::shared_ptr<ColumnWriter> ColumnWriter::Make(ColumnChunkMetaDataBuilder* met
                               descr->physical_type() != Type::BOOLEAN;
   Encoding::type encoding = properties->encoding(descr->path());
   if (encoding == Encoding::UNKNOWN) {
-    encoding = (descr->physical_type() == Type::BOOLEAN &&
-                properties->version() != ParquetVersion::PARQUET_1_0 &&
-                properties->data_page_version() == ParquetDataPageVersion::V2)
-                   ? Encoding::RLE
-                   : Encoding::PLAIN;
+    encoding = ChooseFallbackEncoding(descr->physical_type(), properties->version(),
```

Comment:
nit: rename it to `ChooseNonDictEncoding`. It would be weird to see `fallback` here.

Reply:
Sure. Done.